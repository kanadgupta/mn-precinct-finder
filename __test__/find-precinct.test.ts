import { describe, expect, it } from 'vitest';

import findPrecinct from '../src/lib/find-precinct.js';

describe('#findPrecinct', () => {
  it('should return precinct data if passed as array', () => {
    const precinct = findPrecinct([-93.265, 44.9778]);
    expect(precinct).toMatchSnapshot();
  });

  it('should return precinct data if passed as geojson coordinate', () => {
    const precinct = findPrecinct({ type: 'Point', coordinates: [-93.265, 44.9778] });
    expect(precinct).toMatchSnapshot();
  });

  it('should throw error if non-Minnesota coordinates are passed', () => {
    expect(() => findPrecinct({ type: 'Point', coordinates: [0, 0] })).toThrow('Unable to find a precinct match');
  });

  it('should throw error if total garbage is passed', () => {
    // @ts-expect-error deliberately passing invalid data
    expect(() => findPrecinct(['yo'])).toThrow('coordinates must be at least 2 numbers long');
  });
});
