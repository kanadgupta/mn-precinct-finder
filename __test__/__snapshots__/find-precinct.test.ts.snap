// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`#findPrecinct > should return precinct data if passed as array 1`] = `
{
  "CongDist": "5",
  "County": "Hennepin",
  "CountyID": "27",
  "CtyComDist": "03",
  "Hospital": "no data",
  "Judicial": "04",
  "MCDCode": "135",
  "MCDName": "Minneapolis",
  "MNLegDist": "59B",
  "MNSenDist": "59",
  "Park": "4",
  "Precinct": "Minneapolis W-7 P-11",
  "PrecinctCode": "1705",
  "PrecinctID": "270531705",
  "SoilAndWater": "no data",
  "Ward": "W-07",
}
`;

exports[`#findPrecinct > should return precinct data if passed as geojson coordinate 1`] = `
{
  "CongDist": "5",
  "County": "Hennepin",
  "CountyID": "27",
  "CtyComDist": "03",
  "Hospital": "no data",
  "Judicial": "04",
  "MCDCode": "135",
  "MCDName": "Minneapolis",
  "MNLegDist": "59B",
  "MNSenDist": "59",
  "Park": "4",
  "Precinct": "Minneapolis W-7 P-11",
  "PrecinctCode": "1705",
  "PrecinctID": "270531705",
  "SoilAndWater": "no data",
  "Ward": "W-07",
}
`;
